fdbserver/TLogServer.actor.cpp

```diff
@@ -2770,6 +2791,16 @@ ACTOR Future<Void> tLog( IKeyValueStore* persistentData, IDiskQueue* persistentQ
 					}
 				}
 				when ( wait( error ) ) { throw internal_error(); }
+				when ( wait( activeSharedTLog->onChange() ) ) {
```

Comment:
There is a delay before we get to this onChange() call, so if you restore multiple shared logs from disk none of the instances will actually check that they are not the activeSharedTLog

Reply:
Fixed by moving the code out of the choose, and into the loop, and just leaving the `onChange()` as a way to wake up and re-run the loop header.